packages/ai/openai/src/OpenAiEmbeddings.ts

```diff
@@ -0,0 +1,143 @@
+/**
+ * @since 1.0.0
+ */
+import { AiError } from "@effect/ai/AiError"
+import * as Embeddings from "@effect/ai/Embeddings"
+import * as Context from "effect/Context"
+import type * as Duration from "effect/Duration"
+import * as Effect from "effect/Effect"
+import * as Layer from "effect/Layer"
+import type { Simplify } from "effect/Types"
+import type * as Generated from "./Generated.js"
+import { OpenAiClient } from "./OpenAiClient.js"
+
+/**
+ * @since 1.0.0
+ * @category tags
+ */
+export class OpenAiEmbeddingsConfig extends Context.Tag("@effect/ai-openai/OpenAiEmbeddings/Config")<
+  OpenAiEmbeddingsConfig,
+  Simplify<
+    Partial<
+      Omit<
+        typeof Generated.CreateEmbeddingRequest.Encoded,
+        "input"
+      >
+    >
+  >
+>() {
+  /**
+   * @since 1.0.0
+   */
+  static readonly getOrUndefined: Effect.Effect<typeof OpenAiEmbeddingsConfig.Service | undefined> = Effect.map(
+    Effect.context<never>(),
+    (context) => context.unsafeMap.get(OpenAiEmbeddingsConfig.key)
+  )
+}
+
+const makeRequest = (
+  client: OpenAiClient.Service,
+  input: ReadonlyArray<string>,
+  parentConfig: typeof OpenAiEmbeddingsConfig.Service | undefined,
+  options: {
+    readonly model: string
+    readonly maxBatchSize?: number
+    readonly cache?: {
+      readonly capacity: number
+      readonly timeToLive: Duration.DurationInput
+    }
+  }
+) =>
+  Effect.context<never>().pipe(
+    Effect.flatMap((context) => {
+      const localConfig = context.unsafeMap.get(OpenAiEmbeddingsConfig.key)
+      return client.client.createEmbedding({
+        input,
+        model: options.model,
+        ...parentConfig,
+        ...localConfig
+      })
+    }),
+    Effect.map((response) =>
+      response.data.map(({ embedding, index }) => ({
+        embeddings: embedding as Array<number>,
```

Comment:
should we define `EmbeddingVector` data type?

Reply:
I don't see a real reason to do so unless I'm missing something?

I'm just casting from a read only array to a mutable one here.